examples/with-mobx-state-tree-typescript/components/SampleComponent.tsx

```diff
@@ -10,42 +10,27 @@ interface IOwnProps {
   linkTo: string
 }
 
-@inject('store')
-@observer
-class SampleComponent extends React.Component<IOwnProps> {
-  public componentDidMount() {
-    if (!this.props.store) {
-      return
-    }
-    this.props.store.start()
-  }
+const SampleComponent: React.FC<IOwnProps> = observer((props) => {
+  const { lastUpdate, light, start, stop } = useStore('')
 
-  public componentWillUnmount() {
-    if (!this.props.store) {
-      return
+  useEffect(() => {
+    start()
+    return () => {
+      stop()
     }
-    this.props.store.stop()
-  }
+  })
```

Comment:
I assume this should have zero deps to mimic old behavior:
```suggestion
  }, [])
```